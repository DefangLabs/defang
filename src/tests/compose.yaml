name: tests
services:
  dfnx:
    build:
      context: .
      dockerfile: Dockerfile
      args:
        DNS: "${COMPOSE_PROJECT_NAME}-dfnx.prod1.defang.dev"
    env_file:
      - fileName.env
    deploy:
      resources:
        limits:
          cpus: '0.50'
          memory: 512M
        reservations:
          cpus: '0.25'
          memory: 256M
    ports:
      - target: 80
        mode: ingress
      - target: 1234
        # mode: host
    secrets:
      - dummy
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost/"]
      # disable: true

  # dfnx:
  #   build:
  #     context: .
      # dockerfile: Dockerfile.dfn
    # ports:
      # - 80

  echo:
    image: ealen/echo-server
    ports:
      - target: 80
        mode: ingress
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost/"]
    # domainname: echotest.gnafed.click
    profiles:
      - donotstart

secrets:
  dummy:
    external: true
    name: dummy
