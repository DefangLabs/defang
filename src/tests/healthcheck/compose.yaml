services:
  flask1:
    image: flask
    ports:
      - 5000
    healthcheck:
      test: ["CMD", "python", "-c", "import sys, urllib.request;urllib.request.urlopen(sys.argv[1]).read()", "http://localhost/"]
  flask2:
    image: flask
    ports:
      - 5000
    healthcheck:
      test: ["CMD", "python", "-c", "import urllib.request;urllib.request.urlopen('http://127.0.0.1/path').read()"]
  wget:
    image: alpine
    ports:
      - 80
    healthcheck:
      test: ["CMD", "wget", "-q", "--spider", "localhost:80"]
  curl:
    image: curl
    ports:
      - 80
    healthcheck:
      test: ["CMD", "curl", "-f", "localhost"]
  cmd-shell:
    image: alpine
    ports:
      - 5000
    healthcheck:
      test: ["CMD-SHELL", "echo hello", "second line"]
  none:
    image: alpine
    ports:
      - 5000
    healthcheck:
      test: ["NONE", "ignored"]
      interval: 1m
      retries: 3
      timeout: 1s
