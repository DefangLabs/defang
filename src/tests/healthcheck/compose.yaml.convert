[
  {
    "name": "curl",
    "image": "curl",
    "platform": 2,
    "internal": true,
    "healthcheck": {
      "test": [
        "CMD",
        "curl",
        "-f",
        "localhost"
      ]
    },
    "networks": 1
  },
  {
    "name": "flask1",
    "image": "flask",
    "platform": 2,
    "internal": true,
    "healthcheck": {
      "test": [
        "CMD",
        "python",
        "-c",
        "import sys, urllib.request;urllib.request.urlopen(sys.argv[1]).read()",
        "http://localhost/"
      ]
    },
    "networks": 1
  },
  {
    "name": "flask2",
    "image": "flask",
    "platform": 2,
    "internal": true,
    "healthcheck": {
      "test": [
        "CMD",
        "python",
        "-c",
        "import urllib.request;urllib.request.urlopen('http://127.0.0.1/').read()"
      ]
    },
    "networks": 1
  },
  {
    "name": "wget",
    "image": "alpine",
    "platform": 2,
    "internal": true,
    "healthcheck": {
      "test": [
        "CMD",
        "wget",
        "-q",
        "--spider",
        "localhost:5000"
      ]
    },
    "networks": 1
  }
]