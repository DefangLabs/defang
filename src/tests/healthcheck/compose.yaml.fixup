[
  {
    "command": null,
    "entrypoint": null,
    "healthcheck": {
      "test": [
        "CMD-SHELL",
        "echo hello",
        "second line"
      ]
    },
    "image": "alpine",
    "networks": {
      "default": null
    },
    "ports": [
      {
        "mode": "ingress",
        "target": 5000,
        "protocol": "tcp",
        "app_protocol": "http"
      }
    ]
  },
  {
    "command": null,
    "entrypoint": null,
    "healthcheck": {
      "test": [
        "CMD",
        "curl",
        "-f",
        "localhost"
      ]
    },
    "image": "curl",
    "networks": {
      "default": null
    },
    "ports": [
      {
        "mode": "ingress",
        "target": 80,
        "protocol": "tcp",
        "app_protocol": "http"
      }
    ]
  },
  {
    "command": null,
    "entrypoint": null,
    "healthcheck": {
      "test": [
        "CMD",
        "python",
        "-c",
        "import sys, urllib.request;urllib.request.urlopen(sys.argv[1]).read()",
        "http://localhost/"
      ],
      "timeout": "1s"
    },
    "image": "flask",
    "networks": {
      "default": null
    },
    "ports": [
      {
        "mode": "ingress",
        "target": 5000,
        "protocol": "tcp",
        "app_protocol": "http"
      }
    ]
  },
  {
    "command": null,
    "entrypoint": null,
    "healthcheck": {
      "test": [
        "CMD",
        "python",
        "-c",
        "import urllib.request;urllib.request.urlopen('http://127.0.0.1/path').read()"
      ],
      "interval": "10s"
    },
    "image": "flask",
    "networks": {
      "default": null
    },
    "ports": [
      {
        "mode": "ingress",
        "target": 5000,
        "protocol": "tcp",
        "app_protocol": "http"
      }
    ]
  },
  {
    "command": null,
    "entrypoint": null,
    "healthcheck": {
      "test": [
        "NONE",
        "ignored"
      ],
      "timeout": "1s",
      "interval": "1m0s",
      "retries": 3
    },
    "image": "alpine",
    "networks": {
      "default": null
    },
    "ports": [
      {
        "mode": "ingress",
        "target": 5000,
        "protocol": "tcp",
        "app_protocol": "http"
      }
    ]
  },
  {
    "command": null,
    "entrypoint": null,
    "healthcheck": {
      "test": [
        "CMD",
        "wget",
        "-q",
        "--spider",
        "localhost:80"
      ]
    },
    "image": "alpine",
    "networks": {
      "default": null
    },
    "ports": [
      {
        "mode": "ingress",
        "target": 80,
        "protocol": "tcp",
        "app_protocol": "http"
      }
    ]
  }
]