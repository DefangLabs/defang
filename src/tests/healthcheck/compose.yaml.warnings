level=warning msg="service \"curl\": missing compose directive: restart; assuming 'unless-stopped' (add 'restart' to silence)"
level=warning msg="service \"curl\": missing memory reservation; specify deploy.resources.reservations.memory to avoid out-of-memory errors"
level=warning msg="service \"flask1\": missing compose directive: restart; assuming 'unless-stopped' (add 'restart' to silence)"
level=warning msg="service \"flask1\": missing memory reservation; specify deploy.resources.reservations.memory to avoid out-of-memory errors"
level=warning msg="service \"flask2\": missing compose directive: restart; assuming 'unless-stopped' (add 'restart' to silence)"
level=warning msg="service \"flask2\": missing memory reservation; specify deploy.resources.reservations.memory to avoid out-of-memory errors"
level=warning msg="service \"wget\": missing compose directive: restart; assuming 'unless-stopped' (add 'restart' to silence)"
level=warning msg="service \"wget\": missing memory reservation; specify deploy.resources.reservations.memory to avoid out-of-memory errors"