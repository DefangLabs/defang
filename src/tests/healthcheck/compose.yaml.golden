name: healthcheck
services:
  curl:
    healthcheck:
      test:
        - CMD
        - curl
        - -f
        - localhost
    image: curl
    networks:
      default: null
  flask1:
    healthcheck:
      test:
        - CMD
        - python
        - -c
        - import sys, urllib.request;urllib.request.urlopen(sys.argv[1]).read()
        - http://localhost/
    image: flask
    networks:
      default: null
  flask2:
    healthcheck:
      test:
        - CMD
        - python
        - -c
        - import urllib.request;urllib.request.urlopen('http://127.0.0.1/').read()
    image: flask
    networks:
      default: null
  wget:
    healthcheck:
      test:
        - CMD
        - wget
        - -q
        - --spider
        - localhost:5000
    image: alpine
    networks:
      default: null
networks:
  default:
    name: healthcheck_default
